gapic/src/main/com/google/gapid/server/GapitPkgInfoProcess.java

```diff
@@ -72,10 +79,35 @@ protected Exception prepare(ProcessBuilder pb) {
     return null;
   }
 
+  private class SearchingInputStream extends BufferedInputStream {
+    public SearchingInputStream(InputStream in) {
+      super(in);
+    }
+
+    public boolean search(byte[] pattern) throws IOException {
+      byte[] test = new byte[pattern.length];
+      while (true) {
+        mark(pattern.length);
+        if (read(test, 0, pattern.length) < 0) {
+          return false;
+        }
+        if (Arrays.equals(test, pattern)) {
+          return true;
+        }
+        reset();
+        skip(1);
+      }
+    }
+  }
+
   @Override
   protected OutputHandler<PkgInfo.PackageList> createStdoutHandler() {
     return new BinaryHandler<PkgInfo.PackageList>(in -> {
-      byte[] data = ByteStreams.toByteArray(in);
+      SearchingInputStream is = new SearchingInputStream(in);
+      if (!is.search(PACKAGE_DATA_MARKER.getBytes())) {
+        throw new RuntimeException("The gapit command didn't produce the data marker.");
```

Comment:
> The SearchingInputStream class has no additional state, so it's not needed.

Well, it has the buffered reader state. I've wrapped it with a try-with-resources,